spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -186,12 +195,14 @@ abstract class KerasLayer[A <: Activity: ClassTag, B <: Activity: ClassTag, T: C
    * @return node containing current module
    */
   override def inputs(nodes : ModuleNode[T]*): ModuleNode[T] = {
-    Util.excludeNotKeras(nodes.map(_.element))
+    excludeInvalidLayers(nodes.map(_.element))
+    ensureNotShared(nodes.map{_.element})
     if (!nodes.isEmpty) { // as there's  Identity().inputs() within Graph
     val inputShape = Shape(nodes.map{_.element.getOutputShape()}.toList)
       this.build(inputShape)
     }
-    super.inputs(nodes: _*)
+
+    processInputs(nodes)
```

Comment:
this can be replaced with super.inputs() only that the `excludeInvalidLayers` would be executed twice